cpg-concepts/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/concepts/Extensions.kt

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2025, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+@file:Suppress("CONTEXT_RECEIVERS_DEPRECATED")
+
+package de.fraunhofer.aisec.cpg.passes.concepts
+
+import de.fraunhofer.aisec.cpg.graph.Node
+import de.fraunhofer.aisec.cpg.graph.OverlayNode
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.CallExpression
+import kotlin.reflect.KClass
+
+/**
+ * The core DSL function of our tagging API. It represents a configuration-style DSL to define a
+ * tagging or mapping between nodes in the graph and desired "overlay nodes". It only represents a
+ * configuration (in the form of the [TaggingContext] class) -- the actual assignment is done
+ * independently, e.g., by the [TagOverlaysPass], which takes the context as an input.
+ *
+ * The [body] of this function is a [TaggingContext] that allows for further specification of which
+ * nodes are "tagged" with [OverlayNode]s. Currently, the only allowed statements are:
+ * - [each]: Applies the specified overlay nodes to "each" of the nodes selected.
+ *
+ * In the following example, each [CallExpression] with the name "foo" is tagged with an overlay
+ * class `Bar`:
+ * ```Kotlin
+ * tag {
+ *   each<CallExpression>("foo").with {
+ *     Bar()
+ *   }
+ * }
+ * ```
+ *
+ * The [TaggingContext] is also returned, which can then be used by the [TagOverlaysPass].
+ */
+fun tag(body: TaggingContext.() -> Unit): TaggingContext {
+    val ctx = TaggingContext()
+    body(ctx)
+    return ctx
+}
+
+/**
+ * each: Applies the specified overlay nodes to "each" of the nodes selected. A node is first of all
```

Comment:
Shouldn't this simply be the method name? If so, I wouldn't specify the name as string as refactoring will cause confusion

```suggestion
 * Applies the specified overlay nodes to "each" of the nodes selected. A node is first of all
```